robot/impl/local_robot.go

```diff
@@ -131,6 +131,7 @@ func (r *localRobot) Close(ctx context.Context) error {
 		}
 	}
 	r.activeBackgroundWorkers.Wait()
+	r.sessionManager.Close()
```

Comment:
Moving the `Close` of the session manager to before the close of the resource manager should make the error reported in RSDK-3176 far less likely (`expireLoop` is still async and _could_ encounter a resource not found error that is only representative of resource-manager-close-due-to-server-close). But, it's still nice to have logic in `expireLoop` that doesn't freak out when the session manager's robot is Closing.